GC job cannot reset Schedule after pause Schedule job

**Steps to reproduce the problem:**
1. Setup GC job Schedule
2. Pause all Schedules
3. Reset GC job Schedule
<img width="1426" alt="图片" src="https://user-images.githubusercontent.com/72076317/205874887-6630ed38-9a5d-41c2-a655-dec2e6d874d3.png">


**Versions:**
Please specify the versions of following systems.

- harbor version: v2.7.0-ded7f0d0

**Additional context:**
core.log

```
Dec  6 09:38:54 172.26.0.1 core[1763668]: 2022-12-06T09:38:54Z [DEBUG] [/server/middleware/log/log.go:30]: attach request id 69bedaeb-be57-4119-8194-018c865bd8ca to the logger for the request PUT /api/v2.0/system/gc/schedule
Dec  6 09:38:54 172.26.0.1 core[1763668]: 2022-12-06T09:38:54Z [DEBUG] [/server/middleware/artifactinfo/artifact_info.go:54]: In artifact info middleware, url: /api/v2.0/system/gc/schedule
Dec  6 09:38:54 172.26.0.1 core[1763668]: 2022-12-06T09:38:54Z [DEBUG] [/server/middleware/security/session.go:47][requestID="69bedaeb-be57-4119-8194-018c865bd8ca"]: a session security context generated for request PUT /api/v2.0/system/gc/schedule
Dec  6 09:38:54 172.26.0.1 core[1763668]: 2022-12-06T09:38:54Z [DEBUG] [/pkg/permission/evaluator/admin/admin.go:35]: system administrator admin require update action for resource /system/garbage-collection
Dec  6 09:38:54 172.26.0.1 core[1763668]: 2022-12-06T09:38:54Z [DEBUG] [/pkg/task/task.go:207]: the stop request for task 21004 is sent
Dec  6 09:38:56 172.26.0.1 core[1763668]: 2022-12-06T09:38:56Z [DEBUG] [/server/middleware/log/log.go:30]: attach request id fd0aef1f-6d81-4440-ae63-93bde2b88e90 to the logger for the request GET /api/v2.0/schedules
Dec  6 09:38:56 172.26.0.1 core[1763668]: 2022-12-06T09:38:56Z [DEBUG] [/server/middleware/artifactinfo/artifact_info.go:54]: In artifact info middleware, url: /api/v2.0/schedules?page_size=15&page=1
Dec  6 09:38:56 172.26.0.1 core[1763668]: 2022-12-06T09:38:56Z [DEBUG] [/server/middleware/log/log.go:30]: attach request id e428e351-23aa-4f82-8df6-0f9370442656 to the logger for the request GET /api/v2.0/jobservice/pools/all/workers
Dec  6 09:38:56 172.26.0.1 core[1763668]: 2022-12-06T09:38:56Z [DEBUG] [/server/middleware/artifactinfo/artifact_info.go:54]: In artifact info middleware, url: /api/v2.0/jobservice/pools/all/workers
Dec  6 09:38:56 172.26.0.1 core[1763668]: 2022-12-06T09:38:56Z [DEBUG] [/server/middleware/log/log.go:30]: attach request id 87cfd937-1c9f-492d-8811-918b27f8a73e to the logger for the request GET /api/v2.0/schedules/all/paused
Dec  6 09:38:56 172.26.0.1 core[1763668]: 2022-12-06T09:38:56Z [DEBUG] [/server/middleware/artifactinfo/artifact_info.go:54]: In artifact info middleware, url: /api/v2.0/schedules/all/paused
Dec  6 09:38:56 172.26.0.1 core[1763668]: 2022-12-06T09:38:56Z [DEBUG] [/server/middleware/security/session.go:47][requestID="e428e351-23aa-4f82-8df6-0f9370442656"]: a session security context generated for request GET /api/v2.0/jobservice/pools/all/workers
Dec  6 09:38:56 172.26.0.1 core[1763668]: 2022-12-06T09:38:56Z [DEBUG] [/server/middleware/security/session.go:47][requestID="fd0aef1f-6d81-4440-ae63-93bde2b88e90"]: a session security context generated for request GET /api/v2.0/schedules
Dec  6 09:38:56 172.26.0.1 core[1763668]: 2022-12-06T09:38:56Z [DEBUG] [/pkg/permission/evaluator/admin/admin.go:35]: system administrator admin require list action for resource /system/jobservice-monitor
Dec  6 09:38:56 172.26.0.1 core[1763668]: 2022-12-06T09:38:56Z [DEBUG] [/pkg/permission/evaluator/admin/admin.go:35]: system administrator admin require list action for resource /system/jobservice-monitor
Dec  6 09:38:56 172.26.0.1 core[1763668]: 2022-12-06T09:38:56Z [DEBUG] [/server/middleware/log/log.go:30]: attach request id b973ef25-e140-49e9-8c6f-0749c3c64f97 to the logger for the request GET /api/v2.0/jobservice/queues
Dec  6 09:38:56 172.26.0.1 core[1763668]: 2022-12-06T09:38:56Z [DEBUG] [/server/middleware/artifactinfo/artifact_info.go:54]: In artifact info middleware, url: /api/v2.0/jobservice/queues
Dec  6 09:38:56 172.26.0.1 core[1763668]: 2022-12-06T09:38:56Z [DEBUG] [/server/middleware/security/session.go:47][requestID="87cfd937-1c9f-492d-8811-918b27f8a73e"]: a session security context generated for request GET /api/v2.0/schedules/all/paused
Dec  6 09:38:56 172.26.0.1 core[1763668]: 2022-12-06T09:38:56Z [DEBUG] [/server/middleware/security/session.go:47][requestID="b973ef25-e140-49e9-8c6f-0749c3c64f97"]: a session security context generated for request GET /api/v2.0/jobservice/queues
Dec  6 09:38:56 172.26.0.1 core[1763668]: 2022-12-06T09:38:56Z [DEBUG] [/pkg/permission/evaluator/admin/admin.go:35]: system administrator admin require list action for resource /system/jobservice-monitor
Dec  6 09:39:03 172.26.0.1 core[1763668]: 2022-12-06T09:39:03Z [DEBUG] [/server/middleware/log/log.go:30]: attach request id ceb9c888-5637-4bde-88a1-0d7499ba6cae to the logger for the request GET /api/v2.0/replication/executions
Dec  6 09:39:03 172.26.0.1 core[1763668]: 2022-12-06T09:39:03Z [DEBUG] [/server/middleware/artifactinfo/artifact_info.go:54]: In artifact info middleware, url: /api/v2.0/replication/executions?policy_id=1&page_size=15&page=1
Dec  6 09:39:03 172.26.0.1 core[1763668]: 2022-12-06T09:39:03Z [DEBUG] [/server/middleware/security/session.go:47][requestID="ceb9c888-5637-4bde-88a1-0d7499ba6cae"]: a session security context generated for request GET /api/v2.0/replication/executions
Dec  6 09:39:03 172.26.0.1 core[1763668]: 2022-12-06T09:39:03Z [DEBUG] [/pkg/permission/evaluator/admin/admin.go:35]: system administrator admin require list action for resource /system/replication
Dec  6 09:39:04 172.26.0.1 core[1763668]: 2022-12-06T09:39:04Z [ERROR] [/lib/http/error.go:56]: {"errors":[{"code":"UNKNOWN","message":"unknown: stopping the execution 7 timeout"}]} 
Dec  6 09:39:06 172.26.0.1 core[1763668]: 2022-12-06T09:39:06Z [DEBUG] [/server/middleware/log/log.go:30]: attach request id a8891fe0-f236-4386-bd0c-73ae44143b52 to the logger for the request GET /api/v2.0/jobservice/pools/all/workers
Dec  6 09:39:06 172.26.0.1 core[1763668]: 2022-12-06T09:39:06Z [DEBUG] [/server/middleware/artifactinfo/artifact_info.go:54]: In artifact info middleware, url: /api/v2.0/jobservice/pools/all/workers
Dec  6 09:39:06 172.26.0.1 core[1763668]: 2022-12-06T09:39:06Z [DEBUG] [/server/middleware/log/log.go:30]: attach request id d42a7e22-b52f-43c0-adb7-67db4194464b to the logger for the request GET /api/v2.0/schedules/all/paused
Dec  6 09:39:06 172.26.0.1 core[1763668]: 2022-12-06T09:39:06Z [DEBUG] [/server/middleware/artifactinfo/artifact_info.go:54]: In artifact info middleware, url: /api/v2.0/schedules/all/paused
Dec  6 09:39:06 172.26.0.1 core[1763668]: 2022-12-06T09:39:06Z [DEBUG] [/server/middleware/security/session.go:47][requestID="d42a7e22-b52f-43c0-adb7-67db4194464b"]: a session security context generated for request GET /api/v2.0/schedules/all/paused
Dec  6 09:39:06 172.26.0.1 core[1763668]: 2022-12-06T09:39:06Z [DEBUG] [/server/middleware/security/session.go:47][requestID="a8891fe0-f236-4386-bd0c-73ae44143b52"]: a session security context generated for request GET /api/v2.0/jobservice/pools/all/workers
Dec  6 09:39:06 172.26.0.1 core[1763668]: 2022-12-06T09:39:06Z [DEBUG] [/server/middleware/log/log.go:30]: attach request id f61ad9f6-747b-4565-be72-a4dd4d8af824 to the logger for the request GET /api/v2.0/jobservice/queues
Dec  6 09:39:06 172.26.0.1 core[1763668]: 2022-12-06T09:39:06Z [DEBUG] [/server/middleware/log/log.go:30]: attach request id 3309f3af-8d12-42b0-b362-4626a80067a3 to the logger for the request GET /api/v2.0/schedules
Dec  6 09:39:06 172.26.0.1 core[1763668]: 2022-12-06T09:39:06Z [DEBUG] [/server/middleware/artifactinfo/artifact_info.go:54]: In artifact info middleware, url: /api/v2.0/jobservice/queues
Dec  6 09:39:06 172.26.0.1 core[1763668]: 2022-12-06T09:39:06Z [DEBUG] [/server/middleware/artifactinfo/artifact_info.go:54]: In artifact info middleware, url: /api/v2.0/schedules?page_size=15&page=1
Dec  6 09:39:06 172.26.0.1 core[1763668]: 2022-12-06T09:39:06Z [DEBUG] [/pkg/permission/evaluator/admin/admin.go:35]: system administrator admin require list action for resource /system/jobservice-monitor
Dec  6 09:39:06 172.26.0.1 core[1763668]: 2022-12-06T09:39:06Z [DEBUG] [/server/middleware/security/session.go:47][requestID="f61ad9f6-747b-4565-be72-a4dd4d8af824"]: a session security context generated for request GET /api/v2.0/jobservice/queues
Dec  6 09:39:06 172.26.0.1 core[1763668]: 2022-12-06T09:39:06Z [DEBUG] [/server/middleware/security/session.go:47][requestID="3309f3af-8d12-42b0-b362-4626a80067a3"]: a session security context generated for request GET /api/v2.0/schedules
Dec  6 09:39:06 172.26.0.1 core[1763668]: 2022-12-06T09:39:06Z [DEBUG] [/pkg/permission/evaluator/admin/admin.go:35]: system administrator admin require list action for resource /system/jobservice-monitor
Dec  6 09:39:06 172.26.0.1 core[1763668]: 2022-12-06T09:39:06Z [DEBUG] [/pkg/permission/evaluator/admin/admin.go:35]: system administrator admin require list action for resource /system/jobservice-monitor
```

--- a/src/pkg/scheduler/scheduler.go
+++ b/src/pkg/scheduler/scheduler.go
@@ -196,7 +196,13 @@ func (s *scheduler) UnScheduleByID(ctx context.Context, id int64) error {
 		executionID := executions[0].ID
 		// stop the execution
 		if err = s.execMgr.StopAndWait(ctx, executionID, 10*time.Second); err != nil {
-			return err
+			if err == task.ErrTimeOut {
+				// Avoid return this error to the UI, log time out error and continue
+				// the execution will be finally stopped by jobservice
+				log.Debugf("time out when stopping the execution %d, but the execution will be stopped eventually", executionID)
+			} else {
+				return err
+			}
 		}
 		// delete execution
 		if err = s.execMgr.Delete(ctx, executionID); err != nil {

--- a/src/pkg/task/execution.go
+++ b/src/pkg/task/execution.go
@@ -17,7 +17,6 @@ package task
 import (
 	"context"
 	"encoding/json"
-	"fmt"
 	"sync"
 	"time"
 
@@ -34,6 +33,7 @@ var (
 	// ExecMgr is a global execution manager instance
 	ExecMgr               = NewExecutionManager()
 	executionSweeperCount = map[string]uint8{}
+	ErrTimeOut            = errors.New("stopping the execution timeout")
 )
 
 // ExecutionManager manages executions.
@@ -333,7 +333,7 @@ func (e *executionManager) StopAndWait(ctx context.Context, id int64, timeout ti
 		lock.Lock()
 		overtime = true
 		lock.Unlock()
-		return fmt.Errorf("stopping the execution %d timeout", id)
+		return ErrTimeOut
 	case err := <-errChan:
 		return err
 	}

